react/features/chat/components/AbstractChat.js

```diff
@@ -86,6 +92,17 @@ export function _mapDispatchToProps(dispatch: Dispatch<any>) {
          */
         _onSendMessage(text: string) {
             dispatch(sendMessage(text));
+        },
+
+        /**
+         * Recomputes the client size taking chat size into account, such that the content is shifted.
+         *
+         * @param {number} clientWidth - The new width of the window.
+         * @param {number} clientHeight - The new height of the window.
+         * @returns {Function}
+         */
+        _resizeWindow(clientWidth: number, clientHeight: number) {
+            dispatch(clientResized(clientWidth, clientHeight));
```

Comment:
hmm this action is not supposed to be used for that. It is supposed to really store the size of the window.

Reply:
Well from what I've looked into, and also discussed with @saghul this is the main way of recomputing the size of the window in order to properly display the tiles and in this situation, the area in which we want to display the tiles actually has to shrink by the chat width, so it seemed natural to use this method, but I'm open to any other suggestion.

Reply:
I see this is probably the easiest way but we want to use this for a window resizes and to store the window size not the available size for the tile view area.

Proposal:
`setTileViewDimensions` should take into account somehow if the chat is opened or not. Maybe by parameter or you can convert it to async action in order to access the state. You can pass to it the size of the available space for the tiles after calculating it. Please check all the places where we call this method. Also check react/features/filmstrip/middleware.web.js and react/features/filmstrip/subscriber.web.js. You will need to `setTileViewDimensions` when the chat opens or closes like we do for other cases (resize of the window, entering participants, etc).